crates/serde/src/id.rs

```diff
@@ -8,6 +8,11 @@ use serde::{Deserialize, Serialize};
 #[serde(into = "String", try_from = "String")]
 pub struct Id<const DELIM: char>(String);
```

Comment:
I think I would _either_ make this even more generic:

```suggestion
pub struct Id<const DELIM: char, const LOWER: bool>(String);
```

_or_ flatten these all out into separate impls.

:shrug:

Reply:
The even more generic approach works out _really_ nicely.  Thanks!